components/prize_pool/commons/prize_pool.lua

```diff
@@ -537,6 +544,59 @@ function PrizePool:setLpdbInjector(lpdbInjector)
 	return self
 end
 
+function PrizePool:_storeLpdb()
+	local lpdbTournamentData = {
+		tournament = Variables.varDefault('tournament_name'),
+		parent = Variables.varDefault('tournament_parent'),
+		series = Variables.varDefault('tournament_series'),
+		shortname = Variables.varDefault('tournament_tickername'),
+		startdate = Variables.varDefaultMulti('tournament_startdate', 'tournament_sdate', 'sdate', ''),
+		mode = Variables.varDefault('tournament_mode'),
+		type = Variables.varDefault('tournament_type'),
+		liquipediatier = Variables.varDefault('tournament_liquipediatier'),
+		liquipediatiertype = Variables.varDefault('tournament_liquipediatiertype'),
+		-- publishertier = ,
+		icon = Variables.varDefault('tournament_icon'),
+		icondark = Variables.varDefault('tournament_icondark'),
+		game = Variables.varDefault('tournament_game'),
+	}
+
+	local setWeight = function(lpdbEntry)
+		lpdbEntry.weight = Template.safeExpand(
+			mw.getCurrentFrame(), 'Weight', {
+				math.max(lpdbEntry.prizemoney, 1),
+				lpdbTournamentData.liquipediatier,
+				lpdbEntry.place,
+				lpdbTournamentData.type
+			})
+	end
+
+	local lpdbData = {}
+	for _, placement in ipairs(self.placements) do
+		local lpdbEntries = placement:_getLpdbData()
+
+		Array.forEach(lpdbEntries, function(lpdbEntry) Table.mergeInto(lpdbEntry, lpdbTournamentData) end)
+
+		Array.forEach(lpdbEntries, setWeight)
+
+		Array.extendWith(lpdbData, lpdbEntries)
+	end
+
+	if self._lpdbInjector then
+		for index, lpdbEntry in ipairs(lpdbData) do
+			lpdbData[index] = self._lpdbInjector:adjust(lpdbEntry)
```

Comment:
wouldn't the whole entry information need to be available during the adjusting?

Reply:
Yes. Moved it to its correct place with the information now being properly available.